src/stash.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as fs from 'fs';
+import { ConfigFile, Logger, SfdxError } from '@salesforce/core';
+import { JsonMap, Optional } from '@salesforce/ts-types';
+
+interface StashFile {
+  isGlobal: boolean;
+  filename: string;
+}
+
+export type StashKey = keyof typeof Stash.KEYS;
+
+export interface DeployData extends JsonMap {
+  jobid: string;
+}
+
+export interface MdRetrieveData extends JsonMap {
+  jobid: string;
+  retrievetargetdir: string;
+  zipfilename?: string;
+  unzip?: boolean;
+}
+
+export type StashData = DeployData | MdRetrieveData;
+
+export class Stash {
+  public static KEYS = {
+    MDAPI_DEPLOY: 'MDAPI_DEPLOY',
+    MDAPI_RETRIEVE: 'MDAPI_RETRIEVE',
+    SOURCE_DEPLOY: 'SOURCE_DEPLOY',
+  };
+
+  // singleton instance of the stash file
+  private static instance: ConfigFile<StashFile>;
+
+  private static logger: Logger;
+
+  // A map of Command.id to stash keys
+  private static keyMap = {
```

Comment:
I know you haven't done the UT yet, but I found in mine that the UT approach typically used (extend the command), since it didn't run from the oclif executable, didn't have the id set and I was having to set it.

Think about whether it would make sense to allow the ID to be set when using the stash rather than relying on it being done for every command in every UT that uses stash.  Or maybe that's what we want.  🤷🏻

Reply:
Not sure what you mean.